src/fs.h

```diff
@@ -88,7 +88,7 @@ static inline auto quoted(const std::string& s)
 // Allow safe path append operations.
 static inline path operator+(path p1, path p2)
 {
-    p1 += std::move(p2);
+    p1 += std::move(static_cast<boost::filesystem::path>(p2));
```

Comment:
In commit "fs: Make compatible with boost 1.78" (575ecd0ff5025bdd2f5cb5e52049aaa202245949)

The static_cast creates an unnecessary temporary here and makes the std::move not have any effect because its argument is already temporary. The right way to write this would be `p1 += static_cast<boost::filesystem::path&&>(p2);`

You could also consider using the alternate fix from https://github.com/bitcoin/bitcoin/issues/23846#issuecomment-1000564006. I think that fix is a little better since it gets rid of the nonstandard + operator in cases where we don't need it and also avoids more temporaries.

Reply:
Changed to the correct cast.